frontend/packages/shared/src/hooks/useUniqueKey.ts

```diff
@@ -0,0 +1,38 @@
+import { useRef } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { ArrayUtils } from '@studio/pure-functions';
+
+type UseUniqueKey = {
+  removeKey: (index: number) => void;
+  getUniqueKey: (index: number) => string;
+};
+
+export type UseUniqueKeyArgs = {
+  maxNumberOfKeys: number;
+};
+
+export const useUniqueKey = ({ maxNumberOfKeys }: UseUniqueKeyArgs): UseUniqueKey => {
+  const internalUniqueKeys = useRef<Array<string>>([]);
+
+  const isInternalUniqueKeysOutOfSync = internalUniqueKeys.current.length !== maxNumberOfKeys;
+  if (isInternalUniqueKeysOutOfSync) {
+    internalUniqueKeys.current = [];
+    for (let i = 0; i < maxNumberOfKeys; i++) {
+      const newlyGeneratedKey = uuidv4();
+      internalUniqueKeys.current.push(newlyGeneratedKey);
+    }
+  }
+
+  const getUniqueKey = (index: number): string => {
+    return internalUniqueKeys.current[index];
+  };
+
+  const removeKey = (index: number): void => {
+    ArrayUtils.removeItemByIndex(internalUniqueKeys.current, index);
+  };
+
+  return {
+    getUniqueKey,
+    removeKey,
+  };
```

Comment:
Vi bør vel også returnere en `addKey`-funksjon som utvider listen når det legges til et element? Hvis elementet bare kan legges til på slutten av listen, som i `LogicalExpressionEditor`, er det tilstrekkelig at den funksjonen bare gjør en `push` på den eksisterende listen.

Er det for øvrig noen tanke bak at det noen steder står `uniqueKey`, mens det bare står `key` andre steder?

Reply:
I scenarioet jeg står overfor, blir elementene lagt til på slutten av listen. Selv om dette fungerer fint nå, kan det være fornuftig å implementere en add-metode for å være mer fleksibel i fremtiden. Samtidig kan vi også vente med å utvide funksjonaliteten til behovet faktisk oppstår. Hva tenker du?

Reply:
Ja, det er så klart ikke nødvendig å legge til ting vi ikke trenger. Men vi trenger det i hvert fall hvis vi skal bruke hooken i `LogicalExpressionEditor`, så ikke åpne elementer lukker seg når man legger til et nytt.